File: BinaryTree/python/maxPathSum.py
from typing import Optional

from node import Node


def max_path_sum(root: Optional[Node]) -> int:
    if root is None:
        return float("-inf")
    if root.left is None and root.right is None:
        return root.val
    return root.val + max(max_path_sum(root.left), max_path_sum(root.right))

File: BinaryTree/php/maxPathSum.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/TreeNode.php';

function maxPathSum(?TreeNode $root): int
{
    if ($root === null) {
        return PHP_INT_MIN;
    }
    if ($root->left === null && $root->right === null) {
        return $root->val;
    }

    return $root->val + max(maxPathSum($root->left), maxPathSum($root->right));
}
